src/backend/gporca/libgpopt/src/xforms/CXformUpdate2DML.cpp

```diff
@@ -144,13 +144,20 @@ CXformUpdate2DML::Transform(CXformContext *pxfctxt, CXformResult *pxfres,
 	// generate oid column and project operator
 	CExpression *pexprProject = NULL;
 	CColRef *pcrTableOid = NULL;
+	// seems this if statement should be changed, because at adb-6.x-dev plans - there are
+	// there are no info about tableoid, also slot at debug doesn't have this info
```

Comment:
Hmm, this point needs some clarification

Reply:
The logic of generation `oid` request seems strange, but if delete an old logic for partitioned tables (for update operation) and using the `pcrOid` from `LogicalUpdate` will broke some test cases from `CDMLTest` (when update operation doesn't have `pcrOid` info for exapmle replaying of plan from dxl without `OidCol`):
  - src/backend/gporca/data/dxl/minidump/UpdateDistKeyMismatchedDistribution.mdp
  - src/backend/gporca/data/dxl/minidump/UpdateDroppedCols.mdp
  - src/backend/gporca/data/dxl/minidump/UpdateNoDistKeyMismatchedDistribution.mdp
  - src/backend/gporca/data/dxl/minidump/UpdatePartTable.mdp

Here is an assertion (below in code) https://github.com/arenadata/gpdb/blob/762fcb30a9e4a1f0d9af1ff4d9f06fa88e5dc35c/src/backend/gporca/libgpopt/src/xforms/CXformUpdate2DML.cpp#L171
These test doesn't have `OidCol` at `dxlLogicalUpdate` node, so assertion fails.
These tests may be [fixed by adding OidCol](https://github.com/arenadata/gpdb/pull/409/commits/0c670de15ec2c87718b447e785b2a932cff86b9f) to the `dxlLogicalUpdate` node, but seems here may be the next problem: replay of plan generated on previous versions of gpdb (where is no `OidCol` at `dxlLogicalUpdate` node) will fail.
There is two approaches to solve problem:
- remove the `assert` with fixing tests
-  still generate `pcrOid` if `Update` node doesn't have prcOid (this approach seems more suitable, because removing `assert` is not nice)